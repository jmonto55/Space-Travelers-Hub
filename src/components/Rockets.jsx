import { useSelector } from 'react-redux';
import Rocket from './Rocket';

const Rockets = () => {
  const { rocketsList } = useSelector((store) => store.rockets);

  return (
    <section className="flex justify-center items-center px-20 py-5 max-w-7xl m-auto">
      <div className="flex flex-col gap-10">
        {rocketsList.map((rokt) => (
          <Rocket
            key={rokt.id}
            id={rokt.id}
            name={rokt.name}
            description={rokt.description}
            img={rokt.flickr_images[0]}
            reserved={rokt.reserved}
          />
        ))}
      </div>
    </section>
  );
};

export default Rockets;
