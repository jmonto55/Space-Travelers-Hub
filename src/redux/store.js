import { configureStore, combineReducers } from '@reduxjs/toolkit';
import rocketsReducer from './rockets/rocketsSlice';
import missionReducer from './missions/missionsSlice';

const rootReducer = combineReducers({ rockets: rocketsReducer, missions: missionReducer });

const store = configureStore({ reducer: rootReducer });

export default store;
