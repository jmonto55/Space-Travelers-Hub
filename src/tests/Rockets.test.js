import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import Rockets from '../components/Rockets';
import store from '../redux/store';
import '@testing-library/jest-dom/extend-expect';

describe('Test suite for the Rockets component', () => {
  it('Should render a Snapshot of the Rockets component', () => {
    const rockets = render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    expect(rockets).toMatchSnapshot();
  });

  test('Should test if the id Rockets is inside the Rockets component', () => {
    render(
      <Provider store={store}>
        <Rockets />
      </Provider>,
    );
    expect(screen.getByTestId('Rockets')).toBeInTheDocument();
  });
});
