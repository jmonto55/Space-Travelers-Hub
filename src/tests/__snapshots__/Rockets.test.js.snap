// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for the Rockets component Should render a Snapshot of the Rockets component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="flex justify-center items-center px-20 py-5 max-w-screen-2xl m-auto"
        data-testid="Rockets"
      >
        <div
          class="flex flex-col gap-10"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="flex justify-center items-center px-20 py-5 max-w-screen-2xl m-auto"
      data-testid="Rockets"
    >
      <div
        class="flex flex-col gap-10"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
